src/Microsoft.AspNetCore.SignalR.Client/LongPollingTransport.cs

```diff
@@ -0,0 +1,137 @@
+        }
+
+        public void Dispose()
+        {
+            _senderCts.Cancel();
+            _pollCts.Cancel();
+            _pipeline?.Dispose();
+        }
+
+        public Task StartAsync(Uri url, IPipelineConnection pipeline)
+        {
+            _pipeline = pipeline;
+
+            // Schedule shutdown of the poller when the output is closed
+            pipeline.Output.Writing.ContinueWith(_ =>
+            {
+                _pollCts.Cancel();
+                return Task.CompletedTask;
+            });
+
+            // Start sending and polling
+            _sender = SendMessages(Utils.AppendPath(url, "send"), _senderCts.Token);
+            _poller = Poll(Utils.AppendPath(url, "poll"), _pollCts.Token);
+            Running = Task.WhenAll(_sender, _poller);
+
+            return Task.CompletedTask;
+        }
+
+        private async Task Poll(Uri pollUrl, CancellationToken cancellationToken)
+        {
+            try
+            {
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    var request = new HttpRequestMessage(HttpMethod.Get, pollUrl);
+                    request.Headers.UserAgent.Add(DefaultUserAgentHeader);
+                    var response = await _httpClient.SendAsync(request, cancellationToken);
+                    response.EnsureSuccessStatusCode();
+
+                    if (response.StatusCode == HttpStatusCode.NoContent || cancellationToken.IsCancellationRequested)
+                    {
+                        // Transport closed or polling stopped, we're done
+                        break;
+                    }
+                    else
+                    {
+                        // Write the data to the output
+                        var buffer = _pipeline.Output.Alloc();
+                        var stream = new WriteableBufferStream(buffer);
+                        await response.Content.CopyToAsync(stream);
+                        await buffer.FlushAsync();
+                    }
+                }
+
+                // Polling complete
+                _pipeline.Output.Complete();
+            }
+            catch (Exception ex)
+            {
+                // Shut down the output pipeline and log
+                _logger.LogError("Error while polling '{0}': {1}", pollUrl, ex);
+                _pipeline.Output.Complete(ex);
+            }
+        }
+
+        private async Task SendMessages(Uri sendUrl, CancellationToken cancellationToken)
+        {
+            using (cancellationToken.Register(() => _pipeline.Input.Complete()))
+            {
+                try
+                {
+                    while (!cancellationToken.IsCancellationRequested)
+                    {
+                        var result = await _pipeline.Input.ReadAsync();
+                        var buffer = result.Buffer;
+                        if (buffer.Length == 0 || result.IsCompleted)
```

Comment:
Yep, I even used it elsewhere, just hadn't done a pass through this to fix that up :)